src/Raven.Server/Documents/ETL/Providers/Queue/AwsSqs/AwsSqsEtl.cs

```diff
@@ -0,0 +1,270 @@
+                        count += batchMessages.Count;
+                        batchMessages.Clear();
+                    }
+                }
+                catch (Exception ex)
+                {
+                    throw new QueueLoadException($"Failed to prepare message, error reason: '{ex.Message}'", ex);
+                }
+            }
+
+            if (batchMessages.Count > 0)
+            {
+                SendBatchMessages(queueName, batchMessages, queue, idsToDelete, tooLargeDocsErrors);
+                count += batchMessages.Count;
+                batchMessages.Clear();
+            }
+
+            if (tooLargeDocsErrors.Count > 0)
+            {
+                Database.NotificationCenter.EtlNotifications.AddLoadErrors(Tag, Name, tooLargeDocsErrors,
+                    "ETL has partially loaded the data. " +
+                    "Some of the documents were too big (>256KB) to be handled by Aws Sqs. " +
+                    "It caused load errors, that have been skipped. ");
+            }
+        }
+
+        return count;
+    }
+
+    private void SendBatchMessages(string queueName, List<SendMessageBatchRequestEntry> batchMessages,
+        QueueWithItems<AwsSqsItem> queue, List<string> idsToDelete, Queue<EtlErrorInfo> tooLargeDocsErrors)
+    {
+        try
+        {
+            var sendMessageBatchRequest = new SendMessageBatchRequest
+            {
+                QueueUrl = _alreadyCreatedQueues.GetValueOrDefault(queueName),
+                Entries = batchMessages
+            };
+
+            AsyncHelpers.RunSync(() => _queueClient.SendMessageBatchAsync(sendMessageBatchRequest));
+
+            if (queue.DeleteProcessedDocuments)
+            {
+                idsToDelete.AddRange(batchMessages.Select(entry => entry.Id));
+            }
+        }
+        catch (Exception)
+        {
+            // Retry sending one by one if batch fails
+            foreach (var message in batchMessages)
+            {
+                try
+                {
+                    var sendMessageRequest = new SendMessageRequest
+                    {
+                        QueueUrl = _alreadyCreatedQueues.GetValueOrDefault(queueName),
+                        MessageBody = message.MessageBody
+                    };
+
+                    AsyncHelpers.RunSync(() => _queueClient.SendMessageAsync(sendMessageRequest));
+
+                    if (queue.DeleteProcessedDocuments)
+                    {
+                        idsToDelete.Add(message.Id);
+                    }
+                }
+                catch (AmazonSQSException sqsEx)
+                {
+                    if (sqsEx.ErrorCode == "InvalidAttributeValue")
+                    {
+                        tooLargeDocsErrors.Enqueue(new EtlErrorInfo()
+                        {
+                            Date = DateTime.UtcNow, DocumentId = message.Id, Error = sqsEx.Message
```

Comment:
Can a user control message Id? If yes then it might not be true that it's a document id.

Reply:
This Id is required when messages are sent in a batch. The user can not control this Id.
Do you think users should control it? I don't think there is a benefit from it.

Reply:
Do we allow to control it in other queue ETLs? Do we have such concept there?